src/app/client/src/app/modules/player-helper/components/player/player.component.ts

```diff
@@ -66,6 +66,18 @@ export class PlayerComponent implements OnInit, AfterViewInit, OnChanges {
   }
 
   ngOnInit() {
+    // If `sessionStorage` has UTM data; append the UTM data to context.cData
+    if (sessionStorage.getItem('UTM')) {
+      let utmData;
+      try {
+        utmData = JSON.parse(sessionStorage.getItem('UTM'));
+      } catch (error) {
+        throw new Error('JSON Parse Error => UTM data');
+      }
+      if (utmData && _.get(this.playerConfig, 'context.cdata') && !_.isEmpty(this.playerConfig.context.cdata)) {
```

Comment:
Dont check for empty cdata. If it is empty for some use-cases, then also UTM params should be added

Reply:
Updated